File: h5xinfa/src/main/java/com/szty/h5xinfa/DownloadZipDialogManager.java
package com.szty.h5xinfa;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.ProgressBar;
import android.widget.TextView;

import com.afollestad.materialdialogs.MaterialDialog;

import androidx.appcompat.widget.AppCompatButton;

public class DownloadZipDialogManager {

    private android.widget.TextView tvshowprogress;
    private android.widget.ProgressBar progressBar;
    private androidx.appcompat.widget.AppCompatButton btncommit;
    private MaterialDialog materialDialog;

    public DownloadZipDialogManager(Context context) {
        MaterialDialog.Builder builder = new MaterialDialog.Builder(context);
        View view = LayoutInflater.from(context).inflate(R.layout.dialog_download, null);
        this.btncommit = (AppCompatButton) view.findViewById(R.id.btn_commit);
        this.progressBar = (ProgressBar) view.findViewById(R.id.progressBar);
        this.tvshowprogress = (TextView) view.findViewById(R.id.tv_show_progress);
        builder.customView(view, false);
        builder.cancelable(false);
        builder.canceledOnTouchOutside(false);
        progressBar.setMax(100);
        progressBar.setIndeterminate(false);
        //设置不能点击
        btncommit.setEnabled(false);
        btncommit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

            }
        });
        materialDialog = builder.build();
    }
    public void showDialog(){
        if (materialDialog!=null) {
            materialDialog.show();
        }
    }
    public void setProgress(int num){
        progressBar.setProgress(num);
    }
    public void setContent(String content){
        tvshowprogress.setText(content);
    }
    public boolean isShowing(){
        if (materialDialog!=null) {
            return materialDialog.isShowing();
        }else{
            return false;
        }
    }
    public void dismiss(){
        if (materialDialog!=null) {
            materialDialog.dismiss();
        }
    }
}

File: h5xinfa/src/main/java/com/szty/h5xinfa/DownloadZipDialogManager.kt
package com.szty.h5xinfa

import android.content.Context
import android.view.View
import android.widget.ProgressBar
import android.widget.TextView
import androidx.appcompat.widget.AppCompatButton
import com.afollestad.materialdialogs.MaterialDialog
import com.afollestad.materialdialogs.customview.customView
import com.afollestad.materialdialogs.customview.getCustomView

/**
 * @author Ly
 */
class DownloadZipDialogManager(context: Context) {
    private val tvshowprogress: TextView
    private val progressBar: ProgressBar
    private val btncommit: AppCompatButton
    private val materialDialog: MaterialDialog?
    fun showDialog() {
        materialDialog?.show()
    }

    fun setProgress(num: Int) {
        progressBar.progress = num
    }

    fun setContent(content: String?) {
        tvshowprogress.text = content
    }

    val isShowing: Boolean
        get() = materialDialog?.isShowing ?: false

    fun dismiss() {
        materialDialog?.dismiss()
    }

    init {
        materialDialog = MaterialDialog(context)
        materialDialog.customView(R.layout.dialog_download)
        materialDialog.cancelable(false)
        materialDialog.cancelOnTouchOutside(false)
        val view = materialDialog.getCustomView()
        btncommit = view.findViewById<View>(R.id.btn_commit) as AppCompatButton
        progressBar = view.findViewById<View>(R.id.progressBar) as ProgressBar
        tvshowprogress = view.findViewById<View>(R.id.tv_show_progress) as TextView
        progressBar.max = 100
        progressBar.isIndeterminate = false
        //设置不能点击
        btncommit.isEnabled = false
        btncommit.setOnClickListener { }
    }
}